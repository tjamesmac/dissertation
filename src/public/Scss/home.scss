body {
  background-color: #40011E;
  font-family: Arial, Helvetica, sans-serif;
}
.header {
  color: #038C8C;
}
.response {
  color: #038C8C;
}


