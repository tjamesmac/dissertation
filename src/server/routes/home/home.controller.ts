import { Request, Response } from 'express';
import { getWordData, isString  } from '../../helpers/wordProcessing';

// class homeController {
//   path = '/';

//   public postWords = ( req: Request, res: Response ) => {
//     console.log('inside everything');
//     const words = req.body;
//     console.log(words);
//     console.log(words.value);

//     const split = isString(words.value);
//     const getWords = getWordData(split);

//     res.send(getWords);
//   }
//   public getSomething = ( req: Request, res: Response ) => {
//     console.log('i worked');
//   }
// }



export const postHomeController = ( req: Request, res: Response ) => {

  console.log('inside everything');
  const words = req.body;
  console.log(words);
  console.log(words.value);

  const split = isString(words.value);
  const getWords = getWordData(split);

  return res.send(getWords);
};

export default postHomeController;
