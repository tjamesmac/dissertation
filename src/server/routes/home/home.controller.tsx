import { Request, Response } from 'express';
import React from 'react';
import { renderToString } from 'react-dom/server';
import App from '../../../public/Typescript/App/App';
import { getWordData, isString  } from '../../helpers/wordProcessing';
import { htmlTemplate } from '../../htmlTemplate';
import { overAll } from '../../wordnet';

const controller = {
  postHomeController: async ( req: Request, res: Response ) => {
    // console.log('inside everything');
    const words = req.body;
    // this needs to be lowercased before going through the function

    overAll( words.value )
      .then( ( answer ) => {
        for ( const prop of Object.keys(answer) ) {
          if (!answer[prop].length) {
            delete answer[prop];
          }
        }
        return res.send( answer );
      } );
  },
  getSSR: ( req: Request, res: Response ) => {
    const jsx = ( <App /> );
    const reactDom = renderToString( jsx );

    res.writeHead( 200, { 'Content-Type': 'text/html' } );
    res.end( htmlTemplate( reactDom ) );
  },
};

export default controller;
