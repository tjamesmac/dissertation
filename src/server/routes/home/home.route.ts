import express from 'express';
import controller from './home.controller';

const router = express.Router();

router.post('/', controller.postHomeController);
router.get('/', controller.getSSR);

export default router;
