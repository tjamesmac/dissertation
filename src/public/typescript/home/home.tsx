import React from 'react';
import { IHelloProps } from './home.interface';

export const Hello: React.FunctionComponent<{name: string}> = (props: IHelloProps) => {
  return (
    <div className='container'>
      <div className='row'>
        <div className='col-12'>
          <h1 className='header'>Hello, {props.name}. Welcome back!</h1>
        </div>
      </div>
    </div>
  );
};
