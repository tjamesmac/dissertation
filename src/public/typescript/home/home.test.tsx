import { shallow } from 'enzyme';
import * as React from 'react';
import { StaticRouter } from 'react-router';
import { Link } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Home } from './home';

describe('Hello component', () => {
  const name = 'Test';
  it('renders the Hello with a name', () => {
    const result = shallow(<Home />).contains(
      <div className='row'>
      <div className='col-12'>
        <h2>Welcome!</h2>
        <div className='info'>
          <p>
            This program has been built to
            help see whether a job advert contains male or female gendered words.
          </p>
          <p>
            After clicking on the start button below,
            you will be presented with a text input area.
          </p>
          <p>
            You will be able to copy and paste text into
            the input area. After doing so, any words
            that can be edited will be highlighted.
            Hovering over them will allow you to change the word with a synonym that is available.
          </p>
          <p>
            If you decided to modify any of the words manually,
             you can press the check button again to see if there are
             any available words for your edit.
          </p>
          <p>
            When you have submitted the changes to your advert, you will
            be able to choose from two previous posted adverts.
          </p>
          <p>
            Choosing one of the previous adverts will present you with an end page.
          </p>
          <p>
            Thank you for taking part!
          </p>
        </div>
      <Link to='/main' className='btn btn-primary'>Start</Link>
      </div>
    </div>);
    expect(result).toBeTruthy();
  });
  it('it renders correctly', () => {
    const tree = renderer
      .create(
        <StaticRouter>
        <Home />
        </StaticRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
