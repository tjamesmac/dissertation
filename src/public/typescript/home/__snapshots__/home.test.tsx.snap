// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component it renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-12"
  >
    <h2>
      Welcome!
    </h2>
    <div
      className="info"
    >
      <p>
        This program has been built to help see whether a job advert contains male or female gendered words.
      </p>
      <p>
        After clicking on the start button below, you will be presented with a text input area.
      </p>
      <p>
        You will be able to copy and paste text into the input area. After doing so, any words that can be edited will be highlighted. Hovering over them will allow you to change the word with a synonym that is available.
      </p>
      <p>
        If you decided to modify any of the words manually, you can press the check button again to see if there are any available words for your edit.
      </p>
      <p>
        When you have submitted the changes to your advert, you will be able to choose from two previous posted adverts.
      </p>
      <p>
        Choosing one of the previous adverts will present you with an end page.
      </p>
      <p>
        Thank you for taking part!
      </p>
    </div>
    <a
      className="btn btn-primary"
      href="/main"
      onClick={[Function]}
    >
      Start
    </a>
  </div>
</div>
`;
